src/app/backend-api.service.ts

```diff
@@ -921,27 +937,34 @@ export class BackendApiService {
     // create an array of messages to decrypt
     req = req.pipe(
       map((res) => {
-        const encryptedHexes = [];
+        // This array contains encrypted messages with public keys
+        // Public keys of the other party involved in the correspondence
+        const encryptedHexesAndPublicKeys = [];
         for (const threads of res.OrderedContactsWithMessages) {
           for (const message of threads.Messages) {
+            let payload = {
+              EncryptedHex: message.EncryptedText,
+              PublicKey: message.SenderPublicKeyBase58Check,
```

Comment:
Can we either write `PublicKey: message.IsSender? message.RecipientPublicKeyBase58Check : message.SenderPublicKeyBase58Check` or
```
            if (message.IsSender) {
              payload.PublicKey = message.RecipientPublicKeyBase58Check;
            } else {
              payload.PublicKey = message.SenderPublicKeyBase58Check
            }

```